src/ADAL.PCL/AcquireTokenHandlerBase.cs

```diff
@@ -47,7 +47,8 @@ internal abstract class AcquireTokenHandlerBase
         protected AcquireTokenHandlerBase(RequestData requestData)
         {
             this.Authenticator = requestData.Authenticator;
-            this.CallState = CreateCallState(this.Authenticator.CorrelationId);
+            this.CallState = CreateCallState(this.Authenticator.CorrelationId, requestData.RequestId);
+            requestData.CorrelationId = this.CallState.CorrelationId;
```

Comment:
As discussed , instead of requestId now the CorrelationId is now updated into Authenticator. Also its been reset after being consumed for telemetry.